Framework/PythonInterface/plugins/algorithms/CalculateFlux.py

```diff
@@ -41,13 +37,16 @@ def PyInit(self):
 
     def PyExec(self):
         input_ws = self.getProperty('InputWorkspace').value
-        wavelengths = input_ws.extractX().flatten()
-        min_wavelength = np.min(wavelengths)
-        max_wavelength = np.max(wavelengths)
-        blocksize = input_ws.blocksize()
-        width = (max_wavelength-min_wavelength) / blocksize
-        params = [min_wavelength, width, max_wavelength]
-        rebinned = Rebin(InputWorkspace=input_ws, StoreInADS=False, Params=params)
+        if input_ws.getAxis(0).getUnit().unitID() == 'Wavelength' and not input_ws.isCommonBins():
```

Comment:
Regarding the 2, if the unit is dimensionless, it should never be ragged, as this doesn't make sense.

Reply:
Re 2), the scenario I was thinking of was if someone ran this algorithm on a ragged workspace with units of "TOF" for example. Before these changes the workspace would have been rebinned and the call to GroupDetectors would have succeeded (even though the variable names refer to wavelength the Rebin would still work) . After these changes the Rebin is skipped and GroupDetectors would fail I think

So not a problem for your workflow but small chance it might affect someone else. I admit this is a bit obscure though so perhaps not very likely to happen

Reply:
Ok, I see, I switched the condition to be `!='Empty'` instead of `='Wavelength'`. It's only in case of empty units that it must not be ragged.